software/teachee-desktop/src/sample_source/mod.rs

```diff
@@ -0,0 +1,91 @@
+use std::{marker::PhantomData, thread, time::Duration};
+
+use crate::storage::Storage;
+
+mod ft;
+mod sine;
+
+// flatten module structure
+pub use ft::FtSampleSource;
+pub use sine::SineSampleSource;
+
+pub type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+pub enum Channel {
+    VoltageA,
+    VoltageB,
+    CurrentA,
+    CurrentB,
+}
```

Comment:
Wondering if the channels enumerated in this enum should be named in a way more representative of what device they correspond to on the PCB. For example, we may want to indicate in the channel name that the data comes from the high-speed discrete ADC rather than the low-speed built in FPGA ADC. Will need to differentiate between those devices somewhere in the code if not here. The user should probably know when they are using the 1MSPS ADC vs the 40MSPS

Reply:
Don't let this one block you from merging though. We can figure it out later. I don't even have the high-speed ADC up and running yet so not urgent.